### ISSUE TITLE
PostProcessing: Speed up

### ISSUE THREAD

**Issue by Ethan-Russell** [e4st-dev/E4ST.jl#281]
Right now, Postprocessing is slow!!!  Would be great to find a way to speed it up.  Would potentially be quite easy to parallelize `extract_results`, for one.

**Comment by Ethan-Russell** [comment]
reading data: 23 seconds
reading config: 9 seconds
E4STAirModel: 4 seconds
ISRMMap: 3 seconds
E4STAirHealth: 1 second

**Comment by Ethan-Russell** [comment]
```julia
julia> @time compute_welfare(data, :user)
 57.156266 seconds (1.07 G allocations: 19.850 GiB, 6.74% gc time)
```

**Comment by Ethan-Russell** [comment]
After tweaking some type inference stuff, cut the time in roughly half.  I think there is still room for improvement though
```julia
julia> @time compute_result(data, :gen, :cost_of_service_rebate)
 19.772258 seconds (113.68 M allocations: 2.199 GiB, 2.09% gc time, 3.32% compilation time:
 32% of which was recompilation)
julia> @time compute_result(data, :gen, :emis_co2_total, :, :, :)
  0.255609 seconds (1.20 M allocations: 18.247 MiB)
```

**Comment by Ethan-Russell** [comment]
It's a problem with `_getindex` for `Vector{Container}`.  Making a utility to promote these vectors when possible.  Makes things much faster.
```julia
julia> @time compute_result(data, :gen, :emis_co2_total, :, :, :)
  0.002450 seconds (9 allocations: 240 bytes)
7.598690458555555e8
```
That's a roughly 200x speedup!
```julia
julia> @time compute_result(data, :gen, :cost_of_service_rebate)
 17.575054 seconds (106.02 M allocations: 2.060 GiB, 2.04% gc time)
```
This one is still slow because there are some Vector{Container} that can't be promoted, because they have a mixture of `ByYear` and `ByNothing`.
```julia
julia> data = read_processed_results(raw"L:\Project-Gurobi\Workspace3\E4ST_Output\sloan_equity\y2030_no_init\240214_runs_no_init\240214_co2_prc_15");

julia> @time compute_result(data, :gen, :cost_of_service_rebate)
 20.966689 seconds (113.20 M allocations: 2.167 GiB, 6.34% gc time, 0.26% compilation time)

julia> @time promote_cols!(data)
  0.814104 seconds (6.98 M allocations: 249.717 MiB, 14.37% gc time, 28.57% compilation time)

julia> @time compute_result(data, :gen, :cost_of_service_rebate);
  4.589195 seconds (43.74 M allocations: 1.132 GiB, 6.84% gc time)
```
This is acceptable, I think.  `cost_of_service_rebate` is inherently expensive, as it needs to be a custom mod.

**Comment by Ethan-Russell** [comment]
Closed in #282
